### PR TITLE
Fix route alternative thread issue

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Addresses https://github.com/mapbox/mapbox-navigation-android/issues/5119

We also want this in 2.2 and the main branch. But it has conflicts with https://github.com/mapbox/mapbox-navigation-android/pull/5034 that we will resolve after that is merged.

### Changelog

```
<changelog>Trigger RouteAlternativesObserver on main thread</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt [hunk 1]
```diff
@@ -45,6 +52,7 @@ internal class RouteAlternativesController constructor(
         if (observers.isEmpty()) {
             nativeRouteAlternativesController.removeObserver(nativeObserver)
         }
+        mainJobControl.job.cancelChildren()
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt:55]: This should be moved to the if-check above rather than being invoked for each unsubscription.

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt]: Yes! nice catch

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -12,7 +12,7 @@ ext {
   // version which we should use in this build
   def mapboxNavigatorVersion = System.getenv("FORCE_MAPBOX_NAVIGATION_NATIVE_VERSION")
   if (mapboxNavigatorVersion == null || mapboxNavigatorVersion == '') {
-      mapboxNavigatorVersion = '79.0.2'
+      mapboxNavigatorVersion = '79.0.3'
```
Comment (kmadsen) [inline gradle/dependencies.gradle:15]: This is unrelated to "fixing the route alternative thread issue"

But this is an extra change to prepare for the `2.1.0-rc.2` release

Comment (LukasPaczos) [reply gradle/dependencies.gradle:15]: Lets add this to the changelog under "Fixed a downstream issue of offline routing being slow." same as for 2.0.1.

**Review by LukasPaczos — APPROVED** [review]
LGTM
